### PR TITLE
Add Nim string interpolation syntax highlighting

### PR DESCRIPTION
**Pull request by carrexxii**
Screenshot_2024-07-02_23-05-04

### CODE REVIEW

File: plugins/language_nim.lua [hunk 1]
```diff
@@ -88,14 +88,37 @@ for _, type in ipairs(standard_generic_types) do
...
+    syntax  = syntax and {
+      patterns = syntax,
+      symbols  = {},
+    } or nil
+  }
+end
+
+local interpolation_syntax = {
+  { pattern = { '{{', '}}' }, type = "string" },
+  { pattern = { '{', '}', '\\' }, type = "keyword2", syntax = { patterns = patterns, symbols = symbols, } },
```
Comment (Guldoman) [inline plugins/language_nim.lua:104]: Would this be equivalent to:
```suggestion
  { pattern = { '{', '}', '\\' }, type = "keyword2", syntax = ".nim" },
```

File: plugins/language_nim.lua [hunk 2]
```diff
@@ -88,14 +88,37 @@ for _, type in ipairs(standard_generic_types) do
...
   -- strings and chars
-  { pattern = { '"', '"', '\\' },             type = "string" },
-  { pattern = { '"""', '"""[^"]' },           type = "string" },
   { pattern = { "'", "'", '\\' },             type = "literal" },
+  string_pattern('"""', '"""[^"]'),
+  string_pattern('"'  , '"'      ),
+  string_pattern('\\"', '\\"'    ), -- For highlighting strings inside iterpolated blocks
+  -- string interpolation
+  string_pattern('%&"""' , '"""[^"]', interpolation_syntax),
+  string_pattern('fmt"""', '"""[^"]', interpolation_syntax),
```
Comment (Guldoman) [inline plugins/language_nim.lua:119]: `[^"]` is a bit problematic, as it captures the character.
Look into either using Lua frontier patterns or regex with negative lookaheads.

Comment (carrexxii) [reply plugins/language_nim.lua:119]: Ok, I'll have a look. I just kept it from the original syntax

Comment (carrexxii) [reply plugins/language_nim.lua]: I made the change for frontier matches as I understand them and have the same result as in my original image

**Comment by Guldoman** [comment]
Looks good!
Could you also bump the plugin version in `manifest.json`?

**Comment by carrexxii** [comment]
I bumped the version

**Comment by Guldoman** [comment]
Looks good, thank you!

**Pull request merged**